bokeh/io.py

```diff
@@ -311,14 +311,14 @@ def _show_with_state(obj, state, browser, new):
     elif state.server_enabled:
         _show_server_with_state(obj, state, new, controller)
 
-    if state.file:
+    else:
         _show_file_with_state(obj, state, new, controller)
```

Comment:
Though the tests seem to explicitly want this function to be called ... @bryevdv @mattpap thoughts? Note that `_show_file_with_state()` is now able to find a filename if not given via `state`.